src/wp-includes/option.php

```diff
@@ -3214,3 +3214,126 @@ function wp_autoload_values_to_autoload() {
 
 	return array_intersect( $filtered_values, $autoload_values );
 }
+
+
+/**
+ * Filters site metadata for multisite installations.
+ *
+ * This function is specific to multisite setups and provides a way to retrieve
+ * network options. It enforces parameters' constraints and suggests using an alternative function.
+ *
+ * @param mixed $current The value to return if no changes are made.
+ * @param int $object_id Site ID.
+ * @param string $meta_key Metadata key.
+ * @param bool $single Whether to retrieve a single value. Must be true.
+ *
+ * @return mixed The network option value for the specified meta key, or false if the function is used incorrectly.
+ */
+function filter_get_site_metadata( $current, $object_id, $meta_key, $single ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'get_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	if ( ! $single ) {
+		_doing_it_wrong( 'get_metadata', __( 'The $single parameter is not supported in this function.' ), '6.8.0' );
+		return $current;
+	}
+
+	_doing_it_wrong( 'get_metadata', __( 'This function is deprecated. Use get_network_option() instead.' ), '6.8.0' );
```

Comment:
Per my overarching comment, I don't think all those warnings and deprecation messages are needed. We can make this work without them.

Reply:
I really want to push home to developers, that if you are doing this, you are doing this wrong. But I still want the code to work. If developers know what they are doing and don't care, they can always unhook these filters and do use the metadata api as is.

Reply:
I agree with you that _some_ of these warnings make sense. But using the metadata functions for network options is not necessarily wrong. It may not be recommended, but for the most part it works.

There  is only one thing that is wrong: Nobody should be able to store multiple meta values for a single meta key in `sitemeta`, since it's intended for network options, and those are restricted to one value per key. So I think this is the only aspect where developers would be doing something critically wrong and should be warned about.

Reply:
It IS wrong, as caches ( specially if external object caches ) values for options are not updated / created / delete. It also does not run through the existing filters and actions for network options. I think there are serious issues and should be pushing developers to not do this.